lib/python_version_utils.pm

```diff
@@ -77,8 +81,37 @@ sub remove_installed_pythons() {
         my $python_versions = script_output("rpm -q $python3_spec_release | awk -F \'-\' \'{print \$2}\'");
         record_info("Python version", "$python_versions:$default_python");
         next if ($python_versions eq $default_python);
-        zypper_call("remove $python3_spec_release-base");
+        if (is_transactional) {
+            trup_call("pkg rm $python3_spec_release-base");
+            process_reboot(expected_grub => 1, trigger => 1);
+        } else {
+            zypper_call("rm $python3_spec_release-base");
+        }
+    }
+}
+
+sub get_available_fullstack_pythons {
+    # Check if python-rpm-macros is already installed
+    my $is_installed = script_run("rpm -qa | grep -q 'python-rpm-macros'");
```

Comment:
why not a simple `rpm -q python-rpm-macros` ? Returns 0 on success (package found), 1 on failure (package not installed)